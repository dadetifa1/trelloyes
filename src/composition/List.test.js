import React from 'react';
import ReactDOM from 'react-dom';
import ListClass from './List';
import renderer from 'react-test-renderer';


describe('Card component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ListClass />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ListClass header="First list" cards={[{ id: 'a', title: 'First card', content: 'lorem ipsum' }]}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders the UI as expected with no props', () => {
    const tree = renderer
      .create(<ListClass />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
}); 