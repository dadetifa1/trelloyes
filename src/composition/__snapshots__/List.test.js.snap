// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      First list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;

exports[`Card component renders the UI as expected with no props 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      in list padre
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        Default title
      </h3>
      <p>
        Nothing to see there
      </p>
    </div>
  </div>
</section>
`;
